fix(nuxt): delete nuxtApp._runningTransition onResolve

kyumoon opened the pull request:
### 🔗 Linked issue

resolves https://github.com/nuxt/nuxt/issues/33024

### 📚 Description

This PR fixes an issue where nuxtApp._runningTransition was not being reset when navigating between a page with a transition and one without under specific conditions. For more details, please refer to the linked issue.

To resolve this, I've added code to the onResolve hook to properly clear the nuxtApp._runningTransition flag.

danielroe commented:
would you be able to add a runtime/unit regression test for this? (we have some scrollBehavior transition tests you can use as an example)

danielroe commented:
Would you mind updating your branch? I did some work on that test file in https://github.com/nuxt/nuxt/commit/5d783662c36b58e7449b3de00c12e62e38720bc7.

kyumoon commented:
> Thank you! ❤️
> 
> Would you mind updating your branch? I did some work on that test file in [5d78366](https://github.com/nuxt/nuxt/commit/5d783662c36b58e7449b3de00c12e62e38720bc7).

I was a bit confused if you wanted me to update the entire file with your code, so I've gone ahead and done that for now.

kyumoon committed:
fix: delete nuxtApp._runningTransition onResolve

kyumoon committed:
test: verify _runningTransition cleanup after page:loading:end

kyumoon commented:
> @kyumoon the test file was renamed and updated, and an issue I hadn't spotted is that this PR was made against `3.x`.
> 
> * would you rebase this PR against `main` branch?
> * would you add a regression test to `router.options.test.ts` to cover the desired behaviour/fix that this PR adds?

I've changed the target branch to master and added the test code.

danielroe committed:
Merge remote-tracking branch 'origin/main' into fix/scrollToTop

danielroe reviewed (approved):
thank you!
